models/src/test/scala/coop/rchain/models/rholang/SortTest.scala

```diff
@@ -50,6 +50,44 @@ class ScoredTermSpec extends FlatSpec with PropertyChecks with Matchers {
     )
     unsortedTerms.sorted should be(sortedTerms)
   }
+  it should "sort so that whenever scores are unequal then result terms have to be unequal" in {
+    def check[A: Sortable: Arbitrary]: Unit =
+      forAll{ (x: A, y: A) =>
+        whenever(sort(x).score != sort(y).score) {
+          assert(sort(x).term != sort(y).term)
```

Comment:
This only checks half of the equivalence. I'd expect this to read:

```
val xSorted = sort(x)
val ySorted = sort(y)
assert((xSorted.term == ySorted.term) == (xSorted.score == ySorted.score))
```

Please take care to extract duplicate values in generative tests to not make them do the work (e.g. sorting) twice without a need.

Also, please make both the added test cases use the pattern from `"sort so that unequal terms have unequal scores and the other way around"` (just below) so that we're more likely to hit similar terms more often.

Maybe extracting a helper method for that pattern would make sense

Reply:
Done